pkg/http/h2c/h2c.go

```diff
@@ -42,6 +43,6 @@ func ListenAndServe(addr string, h http.Handler) error {
 var DefaultTransport http.RoundTripper = &http2.Transport{
 	AllowHTTP: true,
 	DialTLS: func(netw, addr string, cfg *tls.Config) (net.Conn, error) {
-		return net.Dial(netw, addr)
+		return http.DefaultTransport.(*http.Transport).DialContext(context.Background(), netw, addr)
```

Comment:
This typecast makes me uncomfortable.  I have definitely overwritten `http.DefaultTransport` before to change the default in strange environments.  Can we simply construct our own `http.Transport` here if that's what we want?